components/inputs/docs/form-layout-validation.md

```diff
@@ -0,0 +1,142 @@
+</script>
+<d2l-form>
+  <d2l-input-group>
+    <d2l-input-text label="Name" required style="max-width: 200px;"></d2l-input-text>
+    <d2l-input-fieldset label="Address" label-style="heading">
+      <d2l-input-group>
+        <d2l-input-text label="Street" style="max-width: 350px;"></d2l-input-text>
+        <d2l-input-text label="City" style="max-width: 200px;"></d2l-input-text>
+      </d2l-input-group>
+    </d2l-input-fieldset>
+  </d2l-input-group>
+  <d2l-button primary style="margin-top: 1rem;">Submit</d2l-button>
+</d2l-form>
+```
+
+## Form [d2l-form]
+
+The `<d2l-form>` component is used to wrap inputs which are then validated and submitted together.
+
+Unlike the standard HTML `<form>` element, `<d2l-form>` supports validating and submitting custom elements. It also displays an aggregated summary of validation errors at the top of the form.
+
+<!-- docs: demo code properties name:d2l-form sandboxTitle:'Form' display:block -->
+```html
+<script type="module">
+  import '@brightspace-ui/core/components/button/button.js';
+  import '@brightspace-ui/core/components/form/form.js';
+  import '@brightspace-ui/core/components/inputs/input-group.js';
+  import '@brightspace-ui/core/components/inputs/input-text.js';
+  import '@brightspace-ui/core/components/inputs/input-textarea.js';
+
+  const form = document.querySelector('d2l-form');
+  document.querySelector('d2l-button')
+    .addEventListener('click', () => form.submit());
+  form.addEventListener('d2l-form-submit', e => {
+    const data = e.detail.formData; // data.name, data.description
+  })
+</script>
+<d2l-form>
+  <d2l-input-group>
+    <d2l-input-text label="Name" required style="max-width: 200px;"></d2l-input-text>
+    <d2l-input-textarea label="Description" rows="3" style="max-width: 300px;"></d2l-input-textarea>
+  </d2l-input-group>
+  <d2l-button primary style="margin-top: 1rem;">Submit</d2l-button>
+</d2l-form>
+```
+
+<!-- docs: start hidden content -->
+### Properties
+
+| Property | Type | Description |
+|---|---|---|
+| `no-nesting` | Boolean, default: `false` | Indicates that the form should opt-out of nesting.<br><br>This means that it will not be submitted or validated if an ancestor form is submitted or validated. However, directly submitting or validating a form with `no-nesting` will still trigger submission and validation for its descendant forms unless they also opt-out using `no-nesting`. |
+
+### Events
+- `d2l-form-submit`: Dispatched when the form is submitted. The form data can be obtained from the `detail`'s `formData` property.
+- `d2l-form-invalid`: Dispatched when the form fails validation. The error map can be obtained from the `detail`'s `errors` property.
+- `d2l-form-dirty`: Dispatched whenever any form element fires an `input` or `change` event. Can be used to track whether the form is dirty or not.
+<!-- docs: end hidden content -->
+
+### Submitting and Handling Form Submissions
```

Comment:
I've pared the docs down a fair bit from what used to be in `form.md`. I feel like what's here better captures what consumers need to do and know the vast majority of the time.